clang-tools-extra/clang-tidy/modernize/UseTrailingReturnTypeCheck.cpp

```diff
@@ -384,10 +385,15 @@ void UseTrailingReturnTypeCheck::keepSpecifiers(
 }
 
 void UseTrailingReturnTypeCheck::registerMatchers(MatchFinder *Finder) {
-  auto F = functionDecl(
-               unless(anyOf(hasTrailingReturn(), returns(voidType()),
-                            cxxConversionDecl(), cxxMethodDecl(isImplicit()))))
-               .bind("Func");
+  const auto HasNoWrittenReturnType =
+      anyOf(cxxConversionDecl(), cxxConstructorDecl(), cxxDestructorDecl(),
+            cxxMethodDecl(isImplicit()));
```

Comment:
Note: This check is missing TK_IgnoreUnlessSpelledInSource, in such case those isImplicit woudn't be needed, and check could be faster.